### PR TITLE
Upgrade to the latest version of Octokit

### PR DESCRIPTION
**Pull request by joshmgross**
Supports #133, #155, #179
This updates the `github` authenticated Octokit client available in this action to version 5 of the Octokit client. This change allows users to script using the latest released API.

Since this is a major version update and includes breaking changes, this will come with a v5 release of this action.

The primary breaking change is that REST methods must use `github.rest.*` instead of `github.*`. I've added a note to the README for this change and updated our examples.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,12 +37,12 @@
   "dependencies": {
     "@actions/core": "^1.5.0",
     "@actions/exec": "^1.1.0",
-    "@actions/github": "^4.0.0",
-    "@actions/glob": "^0.1.2",
+    "@actions/github": "^5.0.0",
+    "@actions/glob": "^0.2.0",
```
Comment (hashtagchris) [inline package.json:41]: Why didn't we move to `0.2.0` in https://github.com/actions/github-script/pull/192?

**Review by hashtagchris — APPROVED** [review]
Looks good. Thanks!

File: package.json (hunk 1 above)
Comment (joshmgross) [reply package.json:41]: I ran `npm i @actions/glob` to create #192, which I thought would bump minor versions but I think `0.y.z` has special meaning in semver so it did not bump to the "latest minor version".

>Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

https://semver.org/#spec-item-4

**Review by Echo4F — COMMENTED** [review]
nice work

**Pull request merged**